client-app/components/organisms/table/_internal/table-mobile-item/table-mobile-item.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div
+    class="vc-table-mobile__item"
+    :class="{ 'vc-table-mobile__item_moving': isMoving }"
+    :style="`transform: translateX(${offsetX}px)`"
+    @touchstart="touchStart"
+    @touchmove="touchMove"
+    @touchend="touchEnd"
+    @touchcancel="touchCancel"
+  >
+    <!-- Item left actions -->
+    <TableMobileActions
+      v-if="itemActionsLeft && itemActionsLeft.length"
+      :item="item"
+      :item-actions="itemActionsLeft"
+    ></TableMobileActions>
+
+    <div class="vc-table-mobile__item-content">
+      <!-- Mobile item slot content -->
+      <slot></slot>
+    </div>
+
+    <!-- Item right actions -->
+    <TableMobileActions
+      v-if="itemActionsRight && itemActionsRight.length"
+      :item="item"
+      :item-actions="itemActionsRight"
+    ></TableMobileActions>
+  </div>
+</template>
+
+<script setup lang="ts">
+import TableMobileActions from "../table-mobile-actions/table-mobile-actions.vue";
+import { computed, Ref, ref } from "vue";
+import { ItemAction } from "../../../../types";
+
+const props = defineProps({
+  item: {
+    type: Object,
+    default: () => ({}),
+  },
+
+  actionBuilder: {
+    type: Function,
```

Comment:
which PropType?
`Function as PropType<(obj: unknown) => ItemAction[]>` ?

Why is Builder used instead of a calculated array from props 
e.g. `actions: { type: Array as PropType<ItemAction[]>, ... }` ?

Reply:
@artmakarov 

> Why is Builder used instead of a calculated array from props
e.g. actions: { type: Array as PropType<ItemAction[]>, ... } ?

That was taken from **_platform-manager-sdk_**, maybe because there might be some async operations
@Flanker72